src/Dependencies/Collections/ImmutableSegmentedDictionary`2.cs

```diff
@@ -155,14 +156,18 @@ private ImmutableSegmentedDictionary(SegmentedDictionary<TKey, TValue> dictionar
         public ImmutableSegmentedDictionary<TKey, TValue> AddRange(IEnumerable<KeyValuePair<TKey, TValue>> pairs)
         {
             var self = this;
+            SegmentedDictionary<TKey, TValue>? dictionary = null;
 
             // Optimize the case of adding to an empty collection
-            if (self.IsEmpty && TryCastToImmutableSegmentedDictionary(pairs, out var other) && self.KeyComparer == other.KeyComparer)
+            if (self.IsEmpty)
             {
-                return other;
+                if (TryCastToImmutableSegmentedDictionary(pairs, out var other) && self.KeyComparer == other.KeyComparer)
+                    return other;
+
+                if (pairs.TryGetCount(out var count) && count > 0)
+                    dictionary = new SegmentedDictionary<TKey, TValue>(count, self._dictionary.Comparer);
```

Comment:
Interesting. Can you revert this one change and file a follow-up issue for me to investigate?

Reply:
@sharwell -- still waiting on speedometer for this test insertion, but of the 3 versions of this PR being tested, this is the one I think I like the best. I can obviously remove the change in ImmutableSegmentedDictionary, but then the speedometer results wouldn't really match the change. 

Assuming the speedometer results are promising, would it be agreeable to keep in the change in ImmutableSegmentedDictionary, but for the followup work to be to investigate whether this is the appropriate way to improve this?